tests/phpunit/tests/user/wp-user/getUserBy.php

```diff
@@ -0,0 +1,194 @@
+	protected static $user_id;
+	protected static $user_data = array(
+		'user_login'    => 'user1',
+		'user_nicename' => 'userone',
+		'user_pass'     => 'password',
+		'first_name'    => 'John',
+		'last_name'     => 'Doe',
+		'display_name'  => 'John Doe',
+		'user_email'    => 'jonny@battlefield3.com',
+		'user_url'      => 'http://tacos.com',
+		'role'          => 'contributor',
+		'nickname'      => 'Johnny',
+		'description'   => 'I am a WordPress user that cares about privacy.',
+	);
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$user_id = $factory->user->create( self::$user_data );
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_valid_userid
+	 */
+	public function test_id_field_for_valid_userid( $cast_func ) {
+		$user_id = $cast_func( self::$user_id );
+
+		$user = WP_User::get_data_by( 'id', $user_id );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_id_field_for_valid_userid() {
+		return array(
+			'integer user ID' => array( 'intval' ),
+			'string user ID'  => array( 'strval' ),
+			'float user ID'   => array( 'floatval' ),
+		);
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_invalid_userid
+	 * @group        me
+	 */
+	public function test_id_field_for_invalid_userid( $user_id ) {
+		$this->assertFalse( WP_User::get_data_by( 'id', $user_id ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_id_field_for_invalid_userid() {
+		return array(
+			'integer'             => array( 99999 ),
+			'integer less than 0' => array( -1 ),
+			'integer 0'           => array( 0 ),
+			'non-numeric string'  => array( 'non-numeric' ),
+			'string integer'      => array( '99999' ),
+			'string less than 0'  => array( '-1' ),
+			'string 0'            => array( '0' ),
+		);
+	}
+
+	public function test_alias_of_id() {
+		$user = WP_User::get_data_by( 'ID', self::$user_id );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * @dataProvider data_scalar_value
+	 */
+	public function test_scalar_value( $field, $value ) {
+		if ( 'id' === $field || 'ID' === $field ) {
+			$value = self::$user_id;
+		}
+
+		$user = WP_User::get_data_by( $field, $value );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_scalar_value() {
+		return array(
+			'id'          => array(
+				'field' => 'id',
+				'value' => null,
```

Comment:
I wonder if, given the subject of the test method, the `$value` here and in the next dataset might be better as `'REPL_USER_ID'` for clarity in the test suite.

As `null` is not considered scalar by `is_scalar()` [Docs](https://www.php.net/manual/en/function.is-scalar.php#refsect1-function.is-scalar-description), it initially looked to me as though `null` was being tested in `test_scalar()`. `'REPL_USER_ID'` will continue to be overwritten by `self::$user_id`, so this is just for clarity when reading the tests.